azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -59,6 +63,27 @@ def parse(cls, key):
                 return cls.aad_federated_security
             raise KeyError(key)
 
+        def is_secret(self):
+            """States for each property if it contains secret"""
+            return self in [self.password, self.application_key, self.application_certificate]
+
+        def is_str_type(self):
+            """States whether a word is of type str or not."""
+            return self in [

```

Comment:
would put this as a class var so that:
```python
return self in self.str_props
```

also, given the simple line above, would consider doping this function (condition is easy enough to read without the function)